src/math/vector.lisp

```diff
@@ -31,236 +60,149 @@
-                    (in-list vector2))))
-
-(defmethod add ((vector1 float4) vector2)
-  (make-float4 (map 'vector #'+
-                    (in-list vector1)
-                    (in-list vector2))))
-
-
-(defmethod sub ((vector1 float2) vector2)
-  (make-float2 (map 'vector #'-
-                    (in-list vector1)
-                    (in-list vector2))))
-
-(defmethod sub ((vector1 float3) vector2)
-  (make-float3 (map 'vector #'-
-                    (in-list vector1)
-                    (in-list vector2))))
-
-(defmethod sub ((vector1 float4) vector2)
-  (make-float4 (map 'vector #'-
-                    (in-list vector1)
-                    (in-list vector2))))
-
-
-(defmethod mul ((vector float2) (value real))
-  (make-float2 (map 'vector
-                    (ax:curry #'* value)
-                    (in-list vector))))
-
-(defmethod mul ((vector float3) (value real))
-  (make-float3 (map 'vector
-                    (ax:curry #'* value)
-                    (in-list vector))))
-
-(defmethod mul ((vector float4) (value real))
-  (make-float4 (map 'vector
-                    (ax:curry #'* value)
-                    (in-list vector))))
-
-
-(defmethod mul ((vector float2) (value float2))
-  (make-float2 (map 'vector #'*
-                    (in-list vector)
-                    (in-list value))))
-
-(defmethod mul ((vector float3) (value float3))
-  (make-float3 (map 'vector #'*
-                    (in-list vector)
-                    (in-list value))))
-
-(defmethod mul ((vector float4) (value float4))
-  (make-float4 (map 'vector #'*
-                    (in-list vector)
-                    (in-list value))))
-
-
-(defmethod div ((vector float2) (value real))
-  (make-float2 (map 'vector
-                    (ax:rcurry #'/ value)
-                    (in-list vector))))
-
-(defmethod div ((vector float3) (value real))
-  (make-float3 (map 'vector
-                    (ax:rcurry #'/ value)
-                    (in-list vector))))
-
-(defmethod div ((vector float4) (value real))
-  (make-float4 (map 'vector
-                    (ax:rcurry #'/ value)
-                    (in-list vector))))
-
-
-(defmethod div ((vector float2) (value float2))
-  (make-float2 (map 'vector #'/
-                    (in-list vector)
-                    (in-list value))))
-
-(defmethod div ((vector float3) (value float3))
-  (make-float3 (map 'vector #'/
-                    (in-list vector)
-                    (in-list value))))
-
-(defmethod div ((vector float4) (value float4))
-  (make-float4 (map 'vector #'/
-                    (in-list vector)
-                    (in-list value))))
-
-
-(defmethod dot (vector1 vector2)
+                    :in-vec
+                    (vector a1 a2 a3 a4)))
+    (a1 (make-instance 'float4 :in-vec (vector a1 a1 a1 a1)))))
+
+(defmethod get-size ((vec float2))
+  2)
+
+(defmethod get-size ((vec float3))
+  3)
+
+(defmethod get-size ((vec float4))
+  4)
+
+(defmacro define-vec-op (name result-type map-op)
```

Comment:
If there's a macro for defining operation for vectors, why not also make one for matrices?

Reply:
Because matrices already work, but hey, it seems like a nice enhancement!
Created an issue for that: https://github.com/koto-bank/lbge/issues/90